### PR TITLE
cherry-pick files from feature/tribe-context-propsal branch

### PR DESCRIPTION
**Pull request by lucatume**
A reproduction of https://github.com/moderntribe/tribe-common/pull/922/files on the `bucket/orm` branch.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space after &quot;=&quot;; 0 found | [Context.php:142](https://github.com/moderntribe/tribe-common/blob/009e6cd10b9dd13f165e15fe19a9ca26f890482e/src/Tribe/Context.php#L142)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $title is never used | [Context.php:19](https://github.com/moderntribe/tribe-common/blob/b682998144b48bb74aa06e045751c774b909c81b/src/Tribe/Debug_Bar/Panels/Context.php#L19)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Context.php:89](https://github.com/moderntribe/tribe-common/blob/b682998144b48bb74aa06e045751c774b909c81b/src/Tribe/Debug_Bar/Panels/Context.php#L89)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Context.php [hunk 1]
```diff
@@ -92,41 +212,857 @@ public function is_editing_post( $post_or_type = null ) {
 	 * Helper function to indicate whether the current execution context is AJAX.
 	 *
 	 * This method exists to allow us test code that behaves differently depending on the execution
-	 * context; passing a value to this argument will set it to that value in future checks, a test-related usage.
+	 * context.
 	 *
 	 * @since 4.7.12
-	 *
-	 * @param bool $doing_ajax An injectable status to override the `DOING_AJAX` check.
+	 * @since TBD Removed the $doing_ajax parameter.
```
Comment (sc0ttkclark) [inline src/Tribe/Context.php]: 👏

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $title is never used | [Context.php:19](https://github.com/moderntribe/tribe-common/blob/76545795f6c35b06c70dc04679266e303e317feb/src/Tribe/Debug_Bar/Panels/Context.php#L19)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Context.php:89](https://github.com/moderntribe/tribe-common/blob/76545795f6c35b06c70dc04679266e303e317feb/src/Tribe/Debug_Bar/Panels/Context.php#L89)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $title is never used | [Context.php:19](https://github.com/moderntribe/tribe-common/blob/76545795f6c35b06c70dc04679266e303e317feb/src/Tribe/Debug_Bar/Panels/Context.php#L19)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Context.php:89](https://github.com/moderntribe/tribe-common/blob/76545795f6c35b06c70dc04679266e303e317feb/src/Tribe/Debug_Bar/Panels/Context.php#L89)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $title is never used | [Context.php:19](https://github.com/moderntribe/tribe-common/blob/08a5c4ed1c3453930750a76c0650ab88ee23710b/src/Tribe/Debug_Bar/Panels/Context.php#L19)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Context.php:96](https://github.com/moderntribe/tribe-common/blob/08a5c4ed1c3453930750a76c0650ab88ee23710b/src/Tribe/Debug_Bar/Panels/Context.php#L96)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $title is never used | [Context.php:19](https://github.com/moderntribe/tribe-common/blob/08a5c4ed1c3453930750a76c0650ab88ee23710b/src/Tribe/Debug_Bar/Panels/Context.php#L19)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Context.php:96](https://github.com/moderntribe/tribe-common/blob/08a5c4ed1c3453930750a76c0650ab88ee23710b/src/Tribe/Debug_Bar/Panels/Context.php#L96)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $title is never used | [Context.php:19](https://github.com/moderntribe/tribe-common/blob/43ef3446772783bc7a21876734a3654ddb484345/src/Tribe/Debug_Bar/Panels/Context.php#L19)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Context.php:96](https://github.com/moderntribe/tribe-common/blob/43ef3446772783bc7a21876734a3654ddb484345/src/Tribe/Debug_Bar/Panels/Context.php#L96)

> via jenkins/codesniffer-phpcs

File: src/functions/utils.php [hunk 2]
```diff
@@ -142,19 +142,7 @@ function tribe_exit( $status = '' ) {
...
-		}
-
-		$query_var = Tribe__Utils__Array::get( $_GET, $var );
-
-		if ( null !== $query_var ) {
-			return $query_var;
-		}
-
-		return $default;
+		return Tribe__Utils__Array::get_in_any( array( $_GET, $_POST, $_REQUEST ), $var, $default );
```
Comment (bordoni) [inline src/functions/utils.php]: I feel like we shouldnt use `_REQUEST` here.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $title is never used | [Context.php:19](https://github.com/moderntribe/tribe-common/blob/217178fb4b55a9bd06cb4910f3785a453217acb5/src/Tribe/Debug_Bar/Panels/Context.php#L19)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html' | [Context.php:96](https://github.com/moderntribe/tribe-common/blob/217178fb4b55a9bd06cb4910f3785a453217acb5/src/Tribe/Debug_Bar/Panels/Context.php#L96)

> via jenkins/codesniffer-phpcs

**Pull request merged**